tsdb/db.go

```diff
@@ -1859,6 +1888,11 @@ func (db *DB) EnableCompactions() {
 	level.Info(db.logger).Log("msg", "Compactions enabled")
 }
 
+func (db *DB) generateCompactionDelay() time.Duration {
+	// Up to 10% of the head's chunkRange.
+	return time.Duration(rand.Int63n(db.head.chunkRange.Load()/10)) * time.Millisecond
```

Comment:
is there any technical reason of this 10% limits? It is good already I guess but I am wondering if I would need a bigger delay

Reply:
I believe that the majority of instances should have sufficient resource margins to accommodate a 10% increase in the Head.

In the future, if necessary, we could consider making it customizable.